pandas/core/generic.py

```diff
@@ -8866,9 +8865,7 @@ def mask(
     """
 
     @Appender(_shared_docs["shift"] % _shared_doc_kwargs)
-    def shift(
-        self: FrameOrSeries, periods=1, freq=None, axis=0, fill_value=None
-    ) -> FrameOrSeries:
+    def shift(self, periods=1, freq=None, axis=0, fill_value=None):
```

Comment:
Why was this removed? I think this is correct as is?

Reply:
There's 3 possible returns here.
self.copy()
self.tshift(periods, freq)
self._constructor(new_data).__finalize__(self)

self.copy() has only 1 possible return.
self._constructor(data).__finalize__(self)

self.tshift has 2 possible returns:
self
self._constructor(new_data).__finalize__(self)

making the assumption that self._constructor is restricted to the type of self (maybe should discuss), then this is correct as is.

Series.shift may need to be changed to a typevar at some point so that subclassed Series returns a subclassed Series, but I'm not seeing any mypy errors if this is left as is, so not sure why this is changed?

Reply:
@topper-123 sorry I commented on the wrong one before. Can this not be restored per @simonjayhawkins comment above?

Reply:
Yeah, looks like you're right. I've pushed a new commit.